hyperspy/drawing/signal1d.py

```diff
@@ -125,6 +125,7 @@ def plot(self):
                 # tight_layout is a bit brittle, we do this just in case it
                 # complains
                 pass
+        self._init_draw()
```

Comment:
Wouldn't it be better to simply call ``self.figure.canvas.draw()`` instead?

Reply:
Yes, indeed.